### ISSUE TITLE
Cannot create VPC with only one subnet

### ISSUE THREAD

**Issue by tskiens-quizlet** [terraform-google-modules/terraform-google-network#99]
The VPC module fails when trying to create a VPC with a single Subnet

called via 

```
module "vpc" {
    source  = "terraform-google-modules/network/google"
    version = "~> 1.4.0"

    project_id   = var.project
    network_name = var.network_name
    routing_mode = "GLOBAL"

    subnets = var.vpc_subnets
}
```

wile variables.tf setup as so
```
variable "vpc_subnets" {
  type        = list(map(string))
  default     = [{
      subnet_name = "primary"
      subnet_ip  = "10.99.0.0/24"
      subnet_region = "us-central1"
    },{
      subnet_name = "secondary"
      subnet_ip  = "10.99.1.0/24"
      subnet_region = "us-central1"
    },
  ]
  description = "Name for the subnets"
}
```

the error being produced is
```Error: Error in function call

  on .terraform/modules/vpc/terraform-google-modules-terraform-google-network-7810346/main.tf line 56, in data "google_compute_subnetwork" "created_subnets":
  56:   name       = element(google_compute_subnetwork.subnetwork.*.name, count.index)
    |----------------
    | count.index is 0
    | google_compute_subnetwork.subnetwork is empty tuple

Call to function "element" failed: cannot use element function with an empty
list.


Error: Error in function call

  on .terraform/modules/vpc/terraform-google-modules-terraform-google-network-7810346/main.tf line 57, in data "google_compute_subnetwork" "created_subnets":
  57:   region     = element(google_compute_subnetwork.subnetwork.*.region, count.index)
    |----------------
    | count.index is 0
    | google_compute_subnetwork.subnetwork is empty tuple

Call to function "element" failed: cannot use element function with an empty
list.
```

**Comment by tskiens-quizlet** [comment]
Very  strange that it's claiming the tuple is empty. 

Any help would be thankfully appreciated.

**Comment by aaron-lane** [comment]
Hi @tskiens-quizlet! I apologize for the frustrating experience.

Can you confirm if you are applying with the default value of `vpc_subnets` or if the value is being overridden?

**Comment by tskiens-quizlet** [comment]
Hi Aaron, thanks for the quick reply. 
It seems something was strange in the state file.
As this was still in development and had been applied yet, I was able to `rm -rf  .terraform and .terraform.state` dir and files.

After clearing those paths, I was able to `terraform init` and then `terraform plan` with a single subnet successfully.

Again, thank you for the quick reply! I have no issues currently.
